The goal of this patch is to lower the deployment target for TensorFlow. If I remove macOS 10.13 as the deployment target, the default actually becomes macOS 10.14:
```
tensorflow-swift-apis/Sources/TensorFlow/Core/PythonConversion.swift:17:10: error: compiling for macOS 10.13, but module 'PythonKit' has a minimum deployment target of macOS 10.14: /Users/swiftninjas/s4tf/swift/swift-nightly-install/Library/Developer/Toolchains/swift-tensorflow-DEVELOPMENT-2020-04-29-a.xctoolchain/usr/lib/swift/macosx/PythonKit.swiftmodule/x86_64.swiftmodule
  import PythonKit
         ^
```

I'll try explicitly setting the deployment target to macOS 10.10, to work around TF-1252.

--- a/utils/swift_build_support/swift_build_support/products/pythonkit.py
+++ b/utils/swift_build_support/swift_build_support/products/pythonkit.py
@@ -48,9 +48,12 @@ class PythonKit(product.Product):
             pass
 
         # SWIFT_ENABLE_TENSORFLOW
+        # NOTE(TF-1252): macOS deployment target must be below 10.13 to avoid
+        # libswiftCore.dylib dynamic linker issues on 10.14+ while `tensorflow`
+        # branch contains changes to libswiftCore.dylib.
         target = ''
         if host_target.startswith('macosx'):
-            target = '-DCMAKE_Swift_COMPILER_TARGET=x86_64-apple-macosx10.13'
+            target = '-DCMAKE_Swift_COMPILER_TARGET=x86_64-apple-macosx10.10'
         # SWIFT_ENABLE_TENSORFLOW END
 
         with shell.pushd(self.build_dir):

